dftimewolf/lib/processors/turbinia_base.py

```diff
@@ -1,267 +1,408 @@
+
+    Pre-condition(s): The effective UID can read and write to the temporary
+        directory. file_path must be a path to a valid tgz file containing
+        Turbinia task output files.
 
     Args:
-      timeline_label (str): Label to use to construct the path list.
-      gs_paths (List[str]):  gs:// URI to files that need to be downloaded
-          from GS.
+      tgz_path: File path of a Turbinia task output file (tgz).
+      path_to_collect: A saved path from a Turbinia task.
+
+    Returns:
+      A local path to the extracted file.
+    """
+    local_path = ''
+    if not os.path.exists(tgz_path):
+      self.logger.error(f'File not found {tgz_path}')
+      return local_path
+
+    tempdir = tempfile.mkdtemp()
+    with tarfile.open(tgz_path) as file:
+      members = self._FilterTarMembers(file, path_to_collect)
+      file.extractall(path=tempdir, members=members)
 
-    Raises:
-      TurbiniaException: Upon errors downloading from GCS
+    local_path = os.path.join(tempdir, path_to_collect.lstrip('/'))
+    return local_path
+
+  def DownloadFilesFromAPI(self, task_data: Dict[str, List[str]],
+                           path: str) -> Optional[str]:
+    """Downloads task output data from the Turbinia API server.
+
+    Args:
+      task_data: Response from a /api/request/{request_id} API call.
 
     Returns:
-      list:
-        tuple: containing:
-          str: The timeline label for this path.
-          str: A local path where GS files have been copied to.
+      A local path to Turbinia task output files or None
     """
-    # TODO: Externalize fetching files from GCS buckets to a different module.
-
-    local_paths = []
-    for path in gs_paths:
-      local_path = None
-      output_writer = output_manager.GCSOutputWriter(
-          path, local_output_dir=self._output_path)
-      local_path = output_writer.copy_from(path)
-      self.logger.success('Downloaded {0:s} to {1:s}'.format(path, local_path))
-
-      if local_path:
-        local_paths.append((timeline_label, local_path))
-
-    return local_paths
-
-  def TurbiniaSetUp(self,
-                    project: str,
-                    turbinia_recipe: Union[str, None],
-                    turbinia_zone: str,
-                    sketch_id: int) -> None:
+    result = None
+    api_instance = turbinia_request_results_api.TurbiniaRequestResultsApi(
+        self.client)
+    try:
+      task_id = task_data.get('id')
+      api_response = api_instance.get_task_output(
+          task_id, _preload_content=False)
+      filename = f'{task_id}.tgz'
+
+      # Create a temporary file to write the response to.
+      file = tempfile.NamedTemporaryFile(
+          mode='wb', prefix=f'{filename}', suffix='.tgz', delete=False)
+      local_path = file.name
+      self.logger.info(f'Saving output for task {task_id} to: {local_path}')
+      # Read the response and write to the file.
+      file.write(api_response.read())
```

Comment:
I initially had this as read_chunks which is a generator to avoid memory issues for very large files.  Ran into some issues with that on a private deployment that uses a proxy. Let's chat about this offline.  I think we can use read_chunks here, assuming your concern is around the read() call and large files.